### PR TITLE
feat: post assessment

### PR DESCRIPTION
**Pull request by yogip07**
Closes # 

## ✅ Checks

- [v] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [ ] All the tests have passed

### CODE REVIEW

File: migrations/00004_create_task_schema.sql [hunk 1]
```diff
@@ -26,6 +26,7 @@ CREATE TABLE IF NOT EXISTS user_tasks (
     started_at TIMESTAMPTZ NOT NULL DEFAULT CURRENT_TIMESTAMP,
     finished_at TIMESTAMPTZ NULL,
     satisfaction_level SMALLINT NULL
+    comments VARCHAR(511) NOT NULL DEFAULT ''
```
Comment (aldy505) [inline migrations/00004_create_task_schema.sql]: Jangan gini, bikin migration file baru. Liat https://github.com/pressly/goose?tab=readme-ov-file#create

```sh
cd migrations
goose create user_tasks_comments sql
```

yang line ini di delete aja, di `user_tasks_comments` nanti tinggal `ALTER TABLE ADD COLUMN`

File: task/repository/count_user_task_by_task_id.go [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+func (r *Repository) CountUserTaskByTaskId(ctx context.Context, userId, taskId int64) (total int64, err error) {
+	var countUserTaskSql = `SELECT COUNT(id) AS total FROM user_tasks WHERE  user_id = $1 AND task_id = $2`
+
+	err = r.db.QueryRow(ctx, countUserTaskSql, userId, taskId).Scan(&total)
+	if err != nil {
+		return 0, fmt.Errorf("error executing count query: %w", err)
+	}
+
+	return
+}
```
Comment (aldy505) [inline task/repository/count_user_task_by_task_id.go]: Nggak mau ambil connection dari pool dulu?

File: task/repository/insert_task_assessment.go [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+type InsertTaskAssessmentIn struct {
+	UserId            int64
+	TaskId            int64
+	SatisfactionLevel int64
+	Comments          string
+}
+
+func (r *Repository) InsertTaskAssessment(ctx context.Context, data InsertTaskAssessmentIn) (affected int64, err error) {
+	if data.UserId == 0 || data.TaskId == 0 {
+		return 0, pgx.ErrNoRows
```
Comment (aldy505) [inline task/repository/insert_task_assessment.go]: Jangan return error nya `pgx`, bikin error kita sendiri aja

File: task/service/post_task_assessment.go [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+		}
+	}
+
+	// validate request
+	if err := ValidatePostTaskAssessmentReq(req); err != nil {
+		return &task_stub.EmptyResponse{}, err
+	}
+
+	// insert assessment
+	taskId, _ := strconv.ParseInt(req.TaskId, 10, 64)
```
Comment (aldy505) [inline task/service/post_task_assessment.go]: error nya ini jangan di ignore.

File: task/service/post_task_assessment.go [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+	affected, err := s.taskRepository.InsertTaskAssessment(ctx, repository.InsertTaskAssessmentIn{
+		UserId:            authenticatedUser.ID,
+		TaskId:            taskId,
+		SatisfactionLevel: int64(req.SatisfactionLevel),
+		Comments:          req.Comments,
+	})
+	if err != nil {
+		if errors.Is(err, pgx.ErrNoRows) {
+			return nil, &task_stub.TaskServiceError{
+				StatusCode: http.StatusNotFound,
```
Comment (aldy505) [inline task/service/post_task_assessment.go]: BadRequest aja

File: task/service/post_task_assessment.go [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+			StatusCode: http.StatusInternalServerError,
+			Error:      err,
+		}
+	}
+
+	if affected == 0 {
+		return nil, &task_stub.TaskServiceError{
+			StatusCode: http.StatusNotFound,
+			Error:      fmt.Errorf("task not found"),
+		}
```
Comment (aldy505) [inline task/service/post_task_assessment.go]: Ini nggak usah return error, cuma butuh log somewhere.

**Review by aldy505 — CHANGES_REQUESTED** [review]

File: migrations/00004_create_task_schema.sql (hunk 1 above)
Comment (yogip07) [reply migrations/00004_create_task_schema.sql]: oh kirain gaperlu nambah, soalnya table nya belum dimigrasi

File: task/service/post_task_assessment.go (hunk 4 above)
Comment (yogip07) [reply task/service/post_task_assessment.go]: tetep harus dihandle? meskipun di level validation udah dihandle

File: task/repository/count_user_task_by_task_id.go (hunk 2 above)
Comment (yogip07) [reply task/repository/count_user_task_by_task_id.go]: Bukannya udah dihandle dari pgx nya?
https://github.com/jackc/pgx/blob/6f2ce92356f35263258ef7baa4cde86a3e1c6255/pgxpool/pool.go#L617-L637

File: task/service/post_task_assessment.go (hunk 6 above)
Comment (yogip07) [reply task/service/post_task_assessment.go]: masukin ke log aja?

File: task/repository/count_user_task_by_task_id.go (hunk 2 above)
Comment (aldy505) [reply task/repository/count_user_task_by_task_id.go]: Oh yasudah

Comment (yogip07) [reply task/repository/count_user_task_by_task_id.go]: ok

File: task/repository/insert_task_assessment.go [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+type InsertTaskAssessmentIn struct {
+	UserId            int64
+	TaskId            int64
+	SatisfactionLevel int64
+	Comments          string
+}
+
+func (r *Repository) InsertTaskAssessment(ctx context.Context, data InsertTaskAssessmentIn) (affected int64, err error) {
+	if data.UserId == 0 || data.TaskId == 0 {
+		return 0, task_stub.TaskRepositoryErrNoRows
```
Comment (aldy505) [inline task/repository/insert_task_assessment.go]: jangan ngubah apapun di stub, stub kan di generate sama [proto2http](https://github.com/kodiiing/proto2http), error nya kalo di level repository ya jadi milik repository

File: task/repository/insert_task_assessment.go [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+		AND
+		task_id = $4`
+
+	commandTag, err := r.db.Exec(ctx, insertAssessmentSql,
+		data.SatisfactionLevel, data.Comments,
+		data.UserId, data.TaskId,
+	)
+	if err != nil {
+		if errors.Is(err, pgx.ErrNoRows) {
+			return 0, task_stub.TaskRepositoryErrNoRows
```
Comment (aldy505) [inline task/repository/insert_task_assessment.go]: ini juga sama

**Review by aldy505 — CHANGES_REQUESTED** [review]

File: task/repository/insert_task_assessment.go (hunk 7 above)
Comment (yogip07) [reply task/repository/insert_task_assessment.go]: oh jadi di taruh di repository aja?

**Review by aldy505 — APPROVED** [review]

**Pull request merged**